### PR TITLE
Add V3 document header, signature validation

### PR DESCRIPTION
**Pull request by giarc3**

### CODE REVIEW

File: src/v3/mod.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            iv: iv.try_into().unwrap(),           // Length was validated up-front
+            gcm_tag: gcm_tag.try_into().unwrap(), // Length was validated up-front
+        })
+    }
+}
+
+pub fn verify_signature(key: [u8; AES_KEY_LEN], v3_header: &V3DocumentHeader) -> bool {
+    // If we have no header or authTag, that means this document was encrypted before the header was added, which would only happen if the
+    // document was encrypted with the Java SDK. In that case, we'll just ignore the verification and try to decrypt, which might still work.
+    if v3_header.header.is_none() || !v3_header.has_saas_shield() || v3_header.sig.is_empty() {
```
Comment (coltfred) [inline src/v3/mod.rs:33]: You check below if the header.header is None, seems a bit odd to check it up front as well.

File: src/v3/mod.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                new_gcm_tag == sig.gcm_tag
+            })
+            .unwrap_or(false),
+            _ => false,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (coltfred) [inline src/v3/mod.rs:59]: I'd like to see a test with with no signature and an invalid signature as well since that will exercise the failure conditions.

**Review by coltfred — CHANGES_REQUESTED** [review]
A couple tests I'd like to see.

File: src/v3/mod.rs (hunk 1 above)
Comment (giarc3) [reply src/v3/mod.rs:33]: Removed below

File: src/v3/mod.rs (hunk 2 above)
Comment (giarc3) [reply src/v3/mod.rs:59]: Added

**Review by coltfred — APPROVED** [review]
Thanks!

**Pull request merged**